src/main/java/ay2122s1_cs2103t_w16_2/btbb/logic/parser/util/ParserUtil.java

```diff
@@ -92,6 +95,52 @@ public static Email parseEmail(String email) throws ParseException {
         return new Email(trimmedEmail);
     }
 
+    // Ingredient-level parsers:
+    /**
+     * Parses a {@code String ingredientName} into a {@code IngredientName}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code ingredientName} is invalid.
+     */
+    public static IngredientName parseIngredientName(String ingredientName) throws ParseException {
+        requireNonNull(ingredientName);
+        String trimmedIngredientName = ingredientName.trim();
+        if (!IngredientName.isValidIngredientName(trimmedIngredientName)) {
+            throw new ParseException(IngredientName.MESSAGE_CONSTRAINTS);
+        }
+        return new IngredientName(trimmedIngredientName);
+    }
+
+    /**
+     * Parses a {@code String quantity} into a {@code Quantity}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code quantity} is invalid.
+     */
+    public static Quantity parseQuantity(String quantity) throws ParseException {
+        requireNonNull(quantity);
+        String trimmedQuantity = quantity.trim();
+        if (!Quantity.isValidQuantity(trimmedQuantity)) {
+            throw new ParseException(Quantity.MESSAGE_CONSTRAINTS);
+        }
+        return new Quantity(trimmedQuantity);
```

Comment:
Maybe I could put the try catch block in Quantity's isValidQuantity method instead? since it's handling the input checks.

```
public static boolean isValidQuantity(String test) { 
   Boolean isMatch = test.matches(VALIDATION_REGEX);
   try {
        Integer.parseInt(test);
        return isMatch;
    } catch (NumberFormatException numberFormatException) {
         return false;
    }
}